model_compression_toolkit/core/keras/quantizer/lut_fake_quant.py

```diff
@@ -0,0 +1,118 @@
+from typing import Tuple, Dict, Any, Callable
+
+import numpy as np
+import tensorflow as tf
+from keras.layers import Layer
+from tensorflow.python.util.object_identity import Reference as TFReference
+
+from model_compression_toolkit.core.common.constants import SIGNED, CLUSTER_CENTERS, EPS, \
+    MULTIPLIER_N_BITS, THRESHOLD
+
+
+def activation_lut_kmean_quantizer(activation_n_bits: int,
+                                   quantization_params: Dict[str, np.ndarray]) -> Callable:
+    """
+    Builds a LUT quantizer for layer's activation using the provided params (threshold and clusters).
+    It initiates a fake custom LUT layer that provides the quantizer function.
+
+    Args:
+        activation_n_bits: Number of bits to use for quantization (not used in this function).
+        quantization_params: Dictionary of specific parameters for this quantization function.
+
+    Returns:
+        A fake LUT quantization node.
+    """
+
+    lut_fake_quant = LUTFakeQuant(quantization_params=quantization_params)
+    return lambda x: lut_fake_quant(x)
+
+
+class LUTFakeQuant(Layer):
+    """
+    A custom Keras layer for quantizing activation tensor with non-uniform quantization (using lookup table clustering).
+    """
+
+    def __init__(self, quantization_params: Dict[str, np.ndarray], **kwargs):
+        super(LUTFakeQuant, self).__init__(**kwargs)
+        self.quantization_params = quantization_params
+        self.activation_is_signed = self.quantization_params.get(SIGNED)
+        self.cluster_centers = self.quantization_params.get(CLUSTER_CENTERS)
+        self.threshold = self.quantization_params.get(THRESHOLD)
+
+    def build(self, input_shape: Tuple[int]):
+        """
+        Builds the layer.
+
+        Args:
+            input_shape: The layer's input shape.
+
+        """
+        super(LUTFakeQuant, self).build(input_shape)
+
+    def call(self, input_data: TFReference, **kwargs) -> TFReference:
+        """
+
+        Args:
+            input_data: A Keras input tensor.
+            **kwargs: Optional arguments' dictionary.
+
+        Returns: KerasTensor after applying a non-uniform fake quantization.
+
+        """
+        if self.activation_is_signed is None or self.cluster_centers is None or self.threshold is None:
```

Comment:
I think we should assert it in the init method, instead of returning None. What do you think?

Reply:
This is the behavior in all the quantizers (if the parameters values are None we return None).
If we change it here we need to change it in all the other quantizers.

Reply:
Ok. In general I think it's better to assert, but let's leave it for a different PR